modules/optimization/doc/content/modules/optimization/examples/shapeOpt_Annulus.md

```diff
@@ -0,0 +1,122 @@
+# Constrained Optimization Example: Shape Optimization
+
+## Background
+
+The MOOSE Optimization module provides a flexible framework for solving
+optimization problems in MOOSE. This page will demonstrate constrained shape
+optimization of an annulus while satisfying an equality volume constraint.
+
+# Example: Shape Optimization
+
+## Main-App Optimization
+
+Optimization problems are solved using the [MultiApps](MultiApps/index.md)
+system.  The main application contains the optimization executioner and the
+sub-applications solve the forward and adjoint PDE.   The main application input
+is shown in [main_app].
+
+!listing test/tests/executioners/constrained/shape_optimization/main.i
+         id=main_app
+         caption= Constrained Optimization Main App
+
+For this optimization example, the [GeneralOptimization.md] reporter is used. The
+[!param](/OptimizationReporter/OptimizationReporter/equality_names) option lists
+equality constraints. There is also a
+[!param](/OptimizationReporter/OptimizationReporter/inequality_names), not shown
+below, for listing the inequality constraints. The
+[GeneralOptimization.md] also expects
+the controllable parameters
+[!param](/OptimizationReporter/GeneralOptimization/objective_name) and
+[!param](/OptimizationReporter/GeneralOptimization/num_values_name), which
+are the names of the reporters that will hold the objective value and the number
+of parameters in the forward problem.
+
+
+!listing test/tests/executioners/constrained/shape_optimization/main.i
+         block=OptimizationReporter
+         id=OptRep
+         caption= Main application `OptimizationReporter` block with equality
+         constraints
+
+To enable constrained optimization the
+[!param](/Executioner/Optimize/tao_solver) needs to be set to `taoalmm`, which
+is an Augmented Lagrangian method. This is the only TAO solver in MOOSE that has
+constraint support. The `taoalmm` subsolver is set with [!param](/Executioner/Optimize/petsc_options_iname) and
+[!param](/Executioner/Optimize/petsc_options_value).
+constraint support. This example computes the objective function's gradient using finite differencing in TAO with the `petsc_options_iname` `-tao_fd`.  Finite differencing only uses the forward problem's objective value and does not require an adjoint solve.
+`-tao_almm_subsolver_tao_type` and  `bqnktr` set the subsolver to the Bounded Quasi-Newton-Krylov
+Trust Region Method.
+
+!listing test/tests/executioners/constrained/shape_optimization/main.i
+        block= Executioner
+        id=exec
+        caption= Main application `Executioner` block
+
```

Comment:
Could you add a paragraph and listing describing the `MultiApps` and `Transfers` blocks.